immunopepper/traversal.py

```diff
@@ -589,12 +592,20 @@ def create_output_kmer_cross_samples(output_peptide, k, segm_expr_list, graph_ou
                 sublist_seg = sublist_seg.flatten().tolist()
                 left_past = left
                 W_past = W
-
             # update the cross samples matrix
             if check_database:
                 row_metadata = [kmer_peptide, is_in_junction, junction_annotated, output_peptide.read_frame_annotated]
-                kmer_matrix_edge.add(tuple(row_metadata + list(sublist_jun)))
-                kmer_matrix_segm.add(tuple(row_metadata + list(np.round(sublist_seg, 2))))
+                if is_in_junction:
+                    kmer_matrix_edge.add(tuple(row_metadata + list(sublist_jun)))
+                else:
+                    kmer_matrix_segm.add(tuple(row_metadata + list(np.round(sublist_seg, 2))))
+
+                if len(kmer_matrix_segm) > 1000: # small but dense
+                    save_kmer_matrix(None, kmer_matrix_segm, k, graph_samples, filepointer, out_dir, verbose)
+                    kmer_matrix_segm.clear()
+                if len(kmer_matrix_edge) > 1000: # big but relatively sparse
```

Comment:
Are the comments just as a general statement or do they describe differential treatment of the two arrays? Currently, it looks like they are treated the same way, despite being sparse / dense.